File: frontend/src/models/data/LinearDivider.ts
import DataDivider from "./DataDivider";

export default class LinearDivider extends DataDivider {
    constructor() {
        super();
    }

    evaluate(x: number, y: number): boolean {
        const theta = this.deg * Math.PI / 180 + Math.PI / 4 * 5;
        const sinT = Math.sin(theta);
        const cosT = Math.cos(theta);
        return cosT * (y + this.shiftY) - sinT * (x - this.shiftX) < (sinT * (y + this.shiftY) + cosT * (x - this.shiftX));
    }
}

File: backend/Models/LinearDivider.cs
namespace Graphic_Neural_Network.backend.Models
{
    using Newtonsoft.Json.Linq;
    using System;
    using static System.Net.Mime.MediaTypeNames;

    public class LinearDivider : DataDivider
    {
        public override string Type => "lineardivider";

        public override bool Evaluate(double x, double y)
        {
            double theta = Deg * Math.PI / 180 + Math.PI * 5 / 4;
            double sinT = Math.Sin(theta);
            double cosT = Math.Cos(theta);

            double transformedX = cosT * (y + ShiftY) - sinT * (x - ShiftX);
            double transformedY = sinT * (y + ShiftY) + cosT * (x - ShiftX);

            return transformedX < transformedY;
        }
    }

}
